### DISCUSSION TITLE
Why?

### DISCUSSION THREAD

**Discussion by schwa** [krzyzanowskim/CryptoSwift#982] [Q&A] [answered]
Why not just use CommonCrypto from Swift?

What's the reasoning behind creating this project?

Thanks!

**Comment by krzyzanowskim** [comment, marked as answer]
Because I can.

At first I played with OpenSSL and CommonCrypto, then I moved to pure Swift implementation, partly for fun, partly because I'm about to implement things that are not available in CommonCrypto.

I'm touched that you compared this not to OpenSSL ;)
Cheers.

**Comment by fbartho** [comment]
:+1: Cool. "Because I can" is a great answer. :) It might be worth writing a section in the Readme addressing/expanding on the "why" of this library so people who don't see this issue also get that context :)

**Comment by hpique** [comment]
@schwa @krzyzanowskim Also, because adding CommonCrypto to a Swift framework (not app) is broken.

See http://stackoverflow.com/questions/25248598/importing-commoncrypto-in-a-swift-framework.

**Comment by eneko** [comment]
Thanks for putting this library together. Love #pureswift. Keep the good work!

**Comment by smil2k** [comment]
Very much appreciated! Using "native" libraries invalidates the security benefits of a safe lingo...

**Comment by onmyway133** [comment]
@hpique I think it works now https://github.com/onmyway133/CommonCrypto.swift
But Cryptoswift is so awesome. @krzyzanowskim like in https://github.com/krzyzanowskim/CryptoSwift/blob/master/Sources/CryptoSwift/MD5.swift if the test fails, how do you know where to fix? All seem magic numbers to me
